src/halmos/__main__.py

```diff
@@ -1015,13 +1007,22 @@ class GenModelArgs:
     args: Namespace
     idx: int
     sexpr: str
+    dump_dirname: Optional[str] = None
 
 
 def copy_model(model: Model) -> Dict:
     return {decl: model[decl] for decl in model}
 
 
-def solve(query: str, args: Namespace) -> Tuple[CheckSatResult, Model]:
+def solve(
+    query: str, args: Namespace, dump_filename: Optional[str] = None
+) -> Tuple[CheckSatResult, Model]:
+    if dump_filename:
+        with open(dump_filename, "w") as f:
+            f.write("(set-logic QF_AUFBV)\n")
+            f.write(query)
+            f.write("(get-model)\n")
```

Comment:
`(get-model)` will be an error if the result is unsat.  indeed, z3 provides a command line option that automatically prints a model if sat.  if other smt solvers have such an option, we wouldn't need this.

Reply:
turned out that some smt solvers don't provide such an option, so there is no clear solution that works for all solvers.  for now, we can keep this knowing that the query will generate error when unsat, and come back to this later if needed.

Reply:
sorry forgot to update this: I checked and all the solvers are tried are ok with `(get-model)` even in the unsat case. 

z3 does prints an error, but that's after correctly returning `unsat` so I think it's good overall. 

One thing I still need to check if `(get-model)` makes a big difference in the `sat` case